ros_ws/src/navigation_stack/msgs_converter/src/msgs_converter.cpp

```diff
@@ -0,0 +1,60 @@
+#include "msgs_converter.h"
+#include "car_config.h"
+#include <eigen3/Eigen/Dense>
+
+using namespace Eigen;
+
+MSGSConverter::MSGSConverter()
+{
+    this->m_command_velocity_subscriber =
+        this->m_node_handle.subscribe<geometry_msgs::Twist>(car_config::CMD_VEL, 1, &MSGSConverter::convertCallback,
+                                                            this);
+
+    this->m_to_car_control_publisher =
+        this->m_node_handle.advertise<drive_msgs::drive_param>(car_config::TOPIC_DRIVE_PARAM, 10);
+}
+
+void MSGSConverter::convertCallback(const geometry_msgs::Twist::ConstPtr& cmd_vel_message)
+{
+
+    double velocity_x = cmd_vel_message->linear.x;
+    double velocity_y = cmd_vel_message->linear.y;
+    double velocity_angular = cmd_vel_message->angular.z;
+
+    Vector2d metric_velocity(velocity_x, velocity_y);
+
+    double velocity_result =
+        metric_velocity.norm(); // check if norm is zero, then the steering angle (angle_rad) will be infinity!
+    double erpm_speed = metric_velocity.norm() * car_config::TRANSMISSION / car_config::ERPM_TO_SPEED;
+    double angle_rad = 0;
+
+    if (velocity_angular != 0)
```

Comment:
With the updated cmake file, this creates a warning. Please decide wether it's justified and either remove the comparison or suppress the warning. I think in this case it would be best to do something like `abs(velocity_angluar) < SMALL_NUMBER`.